eng/Directory.Build.Common.props

```diff
@@ -106,6 +106,7 @@
     <GenerateAPIListing Condition="'$(GenerateAPIListing)' == '' AND '$(IsShippingClientLibrary)' == 'true'">true</GenerateAPIListing>
     <EnableSourceLink Condition="'$(EnableSourceLink)' == ''">true</EnableSourceLink>
     <DefineConstants Condition="'$(BuildSnippets)' == 'true'">$(DefineConstants);SNIPPET</DefineConstants>
+    <ApiCompatBaselineTargetFramework Condition="'$(TargetFramework)' == 'net8.0'">netstandard2.0</ApiCompatBaselineTargetFramework>
```

Comment:
Temporarily allow the new targets to fall back to netstandard2.0 for API compat checks

Reply:
Interesting.  I didn't know it was possible to do this!

Reply:
Curious: what motivates doing this in this PR?  Is there a specific issue it is addressing?

Reply:
The API compat tool fails without it, for the net8.0 target it looks for the API  in the most recently released library, which doesn't exist. I looked at old PRs for core when new targets were added and the workaround used was just disabling API compat for that target, but I felt like that would not be ideal to do on a large scale.